projects/wallet-lib-client-for-event-queue/0002-connection-manager-design.md

```diff
@@ -0,0 +1,174 @@
+  - The event data will have the transaction id. // or data
+- `conn_state`
+  - Event data is a `ConnectionState`, so the wallet can know if it is receiving events in real
+    time or not.
+- `reload-data`
+  - When the connection becomes available after it went offline this event will
+    be emitted to signal that the data needs to be reloaded.
+
+#### Methods
+
+- start
+- stop
+- setWalletId
+- getter methods for the configuration
+
+The other methods will be internal.
+
+### FullnodeDataConnection
+
+This is the only connection class not meant to be used by a wallet facade but to
+be used by an external client to get real-time events from the fullnode, an
+example of how to use this is the explorer main page, to update with new
+transactions and blocks.
+
+#### Events
+
+- `network-update`
+  - network events are related to peers of the connected node or when a new tx
+  is accepted by the network.
+- `dashboard`
+  - Dashboard data from the fullnode.
+
+## Changes to the wallet facade
+
+The wallet facade will be updated to use the new connection classes.
+This means that instead of receiving a connection instance the wallet facade
+will receive the connection params and options (i.e. `connection_params`) and
+will instantiate the appropriate connection class.
+
+The wallet-service facade can only use the `WalletServiceConnection`, but the
+`HathorWallet` will need an aditional parameter to choose which connection class
+to use.
+The `connection_mode` argument will be introduced to resolve this issue, it will
+be one of the following:
+
+- `pubsub`
+  - Uses `FullnodePubSubConnection`
+- `event_queue`
+  - Uses `FullnodeEventQueueConnection`
+- `auto`
+
+If not provided, the `pubsub` mode will be used, since this is the most common
+connection used by our wallets.
+
+Ideally the `auto` mode would decide to use either `FullnodePubSubConnection` or
+`FullnodeEventQueueConnection` based on the size of the wallet and other
+parameters, but until we have a better strategy it will simply use
+`FullnodePubSubConnection`.
+
+## FullnodeEventQueueWSConnection connection managementstate
+
+Now that the connection class is responsible for syncing the history, we can
+implement a special logic for the event queue.
+When we are reestablishing the connection we can start from the last
+acknowledged event.
+
+Usually when the connection is lost we will need to clean and sync the history
+of transactions from the beginning since we may have lost some events, but the
+event queue allows us to start streaming events from the moment we lost
+connection.
+This makes it so we don't have to clean the history and can just start the
+stream from where we left off.
+Although this is only applicable when we are connected to the same fullnode, so
+we need to check if the fullnode we are connected to is the same and if it isn't
+we will need to re-sync as usual.
+
+To make sure we are connected to the same fullnode we will use the peer-id, an
```

Comment:
We also need to assert that the `stream_id` is the same from the last sync

Reply:
Nice catch.